### PR TITLE
Connect payment service and apigateway

### PR DESCRIPTION
**Pull request by nonya123456**
Api gateway
- `get /api/v1/user/me/paymentMethods` get all my payment methods
- `post /api/v1/user/me/paymentMethods` add new payment methods
- `patch /api/v1/user/me/paymentMethods` set default payment method
- `post /api/v1/shop/:shopId/bankAccount` add bank account
- call grpc Pay when complete order

Payment service
- implement default payment method
- implement get all my payment methods

### CODE REVIEW

File: apigateway/internal/config/config.go [hunk 1]
```diff
@@ -21,10 +21,11 @@ type ServerConfig struct {
...
 type GRPCConfig struct {
-	User  GrpcConnection `mapstructure:"user"`
-	Auth  GrpcConnection `mapstructure:"auth"`
-	Order GrpcConnection `mapstructure:"order"`
-	Shop  GrpcConnection `mapstructure:"shop"`
+	User    GrpcConnection `mapstructure:"user"`
+	Auth    GrpcConnection `mapstructure:"auth"`
+	Order   GrpcConnection `mapstructure:"order"`
+	Shop    GrpcConnection `mapstructure:"shop"`
+	Payment GrpcConnection `mapstructure:"payment"`
```
Comment (bombnp) [inline apigateway/internal/config/config.go:28]: update config.example.yml too

File: apigateway/internal/v1router/handler/payment_method.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	res, err := h.Grpc.Payment.GetMyPaymentMethods(ctx, &paymentproto.GetMyPaymentMethodsRequest{})
+	if err != nil {
+		return nil, fiber.NewError(fiber.StatusInternalServerError, errors.Wrap(err, "Failed to request GRPC GetMyPaymentMethods").Error())
+	}
+
+	if len(res.PaymentMethods) == 0 {
+		return make([]*paymentproto.PaymentMethod, 0), nil
+	}
```
Comment (bombnp) [inline apigateway/internal/v1router/handler/payment_method.go]: why are we creating a new slice if len is 0? isn't this the exact same thing?

**Review by bombnp — APPROVED** [review]
few changes, otherwise LGTM

just merge after fix

File: apigateway/internal/v1router/handler/payment_method.go (hunk 2 above)
Comment (nonya123456) [reply apigateway/internal/v1router/handler/payment_method.go]: empty slice is sent as nil in grpc.

changed to
```
if res.PaymentMethods == nil {
	return make([]*paymentproto.PaymentMethod, 0), nil
}
```

**Pull request merged**